### PR TITLE
Use seconds for schedule_delay metric

### LINKED ISSUE #18599: datadog parsing error for dagrun.schedule_delay since it is not passed in float type
### Apache Airflow version

2.1.2

### Operating System

Gentoo/Linux

### Versions of Apache Airflow Providers

_No response_

### Deployment

Docker-Compose

### Deployment details

_No response_

### What happened

In datadog-agent logs, got parsing error

[ AGENT ] 2021-09-29 03:20:01 UTC | CORE | ERROR | (pkg/dogstatsd/server.go:411 in errLog) | Dogstatsd: error parsing metric message '"airflow.dagrun.schedule_delay.skew:<Period [2021-09-29T03:20:00+00:00 -> 2021-09-29T03:20:00.968404+00:00]>|ms"': could not parse dogstatsd metric value: strconv.ParseFloat: parsing "<Period [2021-09-29T03:20:00+00:00 -> 2021-09-29T03:20:00.968404+00:00]>": invalid syntax

### What you expected to happen

since datadog agent expects a float, see https://github.com/DataDog/datadog-agent/blob/6830beaeb182faadac40368d9d781b796b4b2c6f/pkg/dogstatsd/parse.go#L119, the schedule_delay should be a float instead of timedelta.

### How to reproduce

_No response_

### Anything else

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by coinzerge**
closes: https://github.com/apache/airflow/issues/18599

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

**Review by potiuk — APPROVED** [review]

**Review by ashb — CHANGES_REQUESTED** [review]
This belongs in the DataDogstatsd adapter inside airflow.stats -- as the normal statsd client accepts timedelta.

I.e. add an isinstance check in here https://github.com/apache/airflow/blob/abdede8eaf4c82555850067d9a327ab519ffe44a/airflow/stats.py#L333-L339

**Comment by uranusjr** [comment]
Agreed, the stat backend should be able to handle timedelta instead.

**Review by uranusjr — APPROVED** [review]

**Comment by coinzerge** [comment]
added test. what do you mean by `we should also add annotations`?

File: airflow/stats.py [hunk 1]
```diff
@@ -331,10 +332,12 @@ def gauge(self, stat, value, rate=1, delta=False, tags=None):
         return None
 
     @validate_stat
-    def timing(self, stat, dt, tags=None):
+    def timing(self, stat: str, dt: Union[float, datetime.timedelta], tags: Optional[List[str]] = None):
```
Comment (uranusjr) [inline airflow/stats.py:335]: Also needed in `StatsLogger` (because that's what's actually gets inspected, not the concrete classes).

**Review by uranusjr — APPROVED** [review]

**Comment by coinzerge** [comment]
@uranusjr anything else i need to do to merge the PR?

**Comment by uranusjr** [comment]
You need to first fix the static checks and tests. Looks like your Union import is breaking things.

**Comment by ashb** [comment]
I'm still not 100% sure this is "right" as many of the timer metrics we track are <1s, so this would just submit 0 values for them making them useless.

**Review by ashb — DISMISSED** [review]
I've just double checked and no, this is _not_ right.

Using the code example from https://docs.datadoghq.com/metrics/dogstatsd_metrics_submission/#timer

```python
from datadog import initialize, statsd
import time
import random

options = {
    'statsd_host':'127.0.0.1',
    'statsd_port':8125
}

initialize(**options)

@statsd.timed('example_metric.timer', tags=["environment:dev,function:my_function"])
def my_function():
  time.sleep(random.randint(0, 10))

my_function()
```

 this is what I see submited to statsd:

```
example_metric.timer:1.836400042520836e-05|ms|#environment:dev_function:my_function
```

**Review by potiuk — APPROVED** [review]

**Comment by kaxil** [comment]
Tests are failing, can you take a look plz

**Comment by eladkal** [comment]
@coinzerge can you please fix the test failure?
`tests/core/test_stats.py::TestDogStats::test_timing: AssertionError: Expected 'timing' to be called once. Called 2 times.`

**Comment by ashb** [comment]
Fixed by #19973

**Pull request closed without merging**